src/alfasim_sdk/alfasim_sdk_api/api.h

```diff
@@ -633,4 +634,123 @@ DLL_EXPORT int get_ucm_fluid_geometrical_properties(
     double D
 );
 
+/*!
+    Gets the current UCM (unit cell model) input data for liquid-liquid flow Pattern
+    calculation. Any available variable by this function is considered for a unit cell,
+    which means that there are variables with one value and there are variables with
+    two values related to the liquid-liquid system (OIL and WATER).
+    If a given variable name is not available, a #NOT_AVAILABLE_DATA error is returned.
+
+    List of `variable_name` with two values (Oil and Water):
+    - `"rho"`: Density [kg/m3]
+    - `"mu"`: Viscosity [Pa.s]
+    - `"U_S"`: Superficial Velocity [m/s]
+
+    It is important to know that the listed `variable_name`s are not available in any phase, only for
+    liquid-liquid systems, in which `OIL` (0, zero) and `WATER` (1, One) are the possible values, and
+    they are available in the #LiquidLiquidSystem enumeration.
+    Because of that, the #error_code must be checked.
+
+    List of `variable_name` with one value:
+    - `"D_h"`: Unit Cell Liquid (Oil-Water) Hydraulic Diameter [m]
+    - `"ks"`: Roughness [m]
+    - `"theta"`: Inclination of the Unit Cell [rad]
+    - `"sigma"`: Liquid-liquid (Oil-Water) Surface Tension [N/m]
+
+    @param[in] ctx ALFAsim's plugins context.
+    @param[out] out Variable value.
+    @param[in] var_name String with the variable name. See the list of possible values above.
+    @param[in] phase_id A #LiquidLiquidSystem value. When the requested variable is not associated
+                            to a phase any value can be passed.
+    @return An #error_code value.
+*/
+DLL_EXPORT int get_ucm_liqliq_flow_pattern_input_variable(
+    void* ctx, double* out, const char* var_name, int phase_id
+);
+
+/*!
+    Gets the current UCM (unit cell model) input data for liquid effective viscosity
+    calculation. Any available variable by this function is considered for a unit cell,
+    which means that there are variables with one value and there are variables with
+    two values related to the liquid-liquid system (OIL and WATER).
+    If a given variable name is not available, a #NOT_AVAILABLE_DATA error is returned.
+
+    List of `variable_name` with two values (Oil and Water):
+    - `"rho"`: Density [kg/m3]
+    - `"mu"`: Viscosity [Pa.s]
+    - `"U_S"`: Superficial Velocity [m/s]
+
+    It is important to know that the listed `variable_name`s are not available in any phase, only for
+    liquid-liquid systems, in which `OIL` (0, zero) and `WATER` (1, One) are the possible values, and
+    they are available in the #LiquidLiquidSystem enumeration.
+    Because of that, the #error_code must be checked.
+
+    List of `variable_name` with one value:
+    - `"alpha_w"` : Water Volume Fraction [kg of `water phase` / kg of `liquid(Oil+Water) phase`]
+    - `"D_h"`: Unit Cell Liquid (Oil-Water) Hydraulic Diameter [m]
+    - `"ks"`: Roughness [m]
```

Comment:
Absolute roughness? Or relative?

Reply:
Thanks, I forgot it. It is Absolute roughness.

- [ ] Absolute Roughness